frontend/packages/client/src/hooks/useStarAnimation.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { useWindowDimensions } from 'hooks';
+
+export default function useStarAnimation({ stars }) {
+  const [starArray, setStarArray] = useState([]);
+  const refsArray = useRef([]);
+  const parentRef = useRef();
+  const { width, height } = useWindowDimensions();
+
+  const addToArrayOfRefes = (currentRef) => {
+    if (!refsArray.current.includes(currentRef)) {
+      refsArray.current.push(currentRef);
+    }
+  };
+
+  const { current: parent } = parentRef;
+
+  useEffect(() => {
+    if (parent) {
+      const maxWidth = parent.clientWidth;
+      const maxHeight = parent.clientHeight;
+      refsArray.current = [];
+      setStarArray(
+        stars.map((position) => ({
+          width: position.width,
+          height: position.height,
+          top: maxHeight * position.topPer,
+          left: maxWidth * position.leftPer,
+        }))
+      );
+    }
+  }, [parent, width, height, stars]);
```

Comment:
added width, height so when resizing stars can change with screen size. Another option would be to provide three arrays ( desktop, tablet, mobile)